aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1107,6 +1107,8 @@ Tensor expand(const Tensor& self, c10::IntArrayRef size, bool /*unused*/) {
            "): the number of sizes provided (", size.size(), ") ",
            "must be greater or equal to the number of dimensions in the tensor (",
            self.dim(), ")");
+  TORCH_CHECK(!self.is_sparse() && !self.is_sparse_csr(), 
```

Comment:
Use:
```suggestion
  TORCH_CHECK(!(self.is_sparse() || at::sparse_csr::is_sparse_compressed(self)), 
```
that will also capture CSC/BSR/BSC layout cases.